### PR TITLE
feat(portal): Update domain parsing logic and tests for parse-domain@8.2.2

### PR DESCRIPTION
**Pull request by Tzal3x**
Handles the case where the `wal.app` is included in the suffix list, thus the logic of parseDomain had to be updated.

- Upgrade parse-domain to 8.2.2 and update node-fetch dependency 
- Add _parseDomain helper and suffix mock test for wal.app edge case 
- Refactor splitUrl to handle wal.app in topLevelDomains 
- Update tests to use new parsing logic and type assertions

### CODE REVIEW

File: portal/common/lib/domain_parsing.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { parseDomain, ParseResultType } from "parse-domain";
+import { parseDomain, ParseResultType, fromUrl, ParseResult} from "parse-domain";
```
Comment (nikos-kitmeridis) [inline portal/common/lib/domain_parsing.ts:4]: `ParseResult` is not used.

File: portal/common/lib/domain_parsing.ts [hunk 2]
```diff
@@ -41,12 +41,18 @@ export function getSubdomainAndPath(url: URL, portalNameLength?: Number): Domain
     {domain: name.wal.app,
     { subdomain: "subname", path: "/index.html"}}
 */
-function splitUrl(url: URL, portalNameLength?: Number): UrlExtract {
-    const parsed = parseDomain(url.hostname);
+export function splitUrl(url: URL, portalNameLength?: Number): UrlExtract {
```
Comment (nikos-kitmeridis) [inline portal/common/lib/domain_parsing.ts:44]: Sinse there is a JSDoc, wanna add a few details for the `portalNameLength` param too?

File: portal/common/lib/domain_parsing.test.ts [hunk 3]
```diff
@@ -24,7 +24,7 @@ const getDomainTestCases: [string, string][] = [
 describe('getDomain', () => {
     getDomainTestCases.forEach(([input, expected]) => {
         test(`${input} -> ${expected}`, () => {
-            const domain = getDomain(new URL(input))
+            const domain = getDomain(new URL(input) as URL)
```
Comment (nikos-kitmeridis) [inline portal/common/lib/domain_parsing.test.ts:27]: For future reference, there's a more idiomatic/native way for parameterized tests.
https://vitest.dev/api/#test-each

File: portal/common/lib/domain_parsing.suffix_mock.test.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { describe, expect, test } from 'vitest'
+import { _parseDomain } from './domain_parsing'
```
Comment (nikos-kitmeridis) [inline portal/common/lib/domain_parsing.suffix_mock.test.ts:5]: Not used/doesn't exist.

File: portal/common/lib/domain_parsing.test.ts (hunk 3 above)
Comment (Tzal3x) [reply portal/common/lib/domain_parsing.test.ts:27]: Ah, nice! I'll edit this thanks 💯

**Review by nikos-kitmeridis — APPROVED** [review]
